src/main/java/io/zeebe/kafka/connect/source/ZeebeSourceTask.java

```diff
@@ -118,8 +117,22 @@ public String version() {
     return VersionInfo.getVersion();
   }
 
+  private void closeIfStopped() {
+    if (taskState.get() == TaskState.STOPPED) {
+      close();
+    }
+  }
+
+  private TaskState closeUnlessPolling(final TaskState currentState) {
+    if (currentState != TaskState.POLLING) {
```

Comment:
Does ths make sense?

Reply:
I mean then you still have one which is in state polling but actually already closed?

Reply:
My understanding here is that `closeUnlessPolling` is called to atomically update the current state. So if we're already polling we don't close anything and transition to stopped, and `poll()` will close everything once it's not polling any more. If we're not polling it's safe to close everything.

Reply:
Aaah yes have overseen this closeIfStopped ok then it makes sense thanks